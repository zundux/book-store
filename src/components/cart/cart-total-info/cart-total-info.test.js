import React from "react";
import CartTotalInfo from "./cart-total-info";

describe("Cart total info", () => {

  test("Cart total info for cart without products", () => {
    const products = [];

    const component = shallow(
      <CartTotalInfo {...{
        products
      }}/>
    );
    matchSnapshot(component);
  });

  test("Cart total info for cart with one product", () => {
    const products = [
      {
        id: 1,
        amount: 1,
        volumeInfo: {
          title: "title",
          imageLinks: {
            smallThumbnail: "img-src"
          },
          saleInfo: {
            listPrice: {
              amount: 1,
              currencyCode: "USD"
            },
            country: "EN"
          },
          authors: ["author 1", "author 2"]
        }
      }
    ];

    const component = shallow(
      <CartTotalInfo {...{
        products
      }}/>
    );
    matchSnapshot(component);
  });

  test("Cart total info for cart with a couple copy of product", () => {
    const products = [
      {
        id: 1,
        amount: 2,
        volumeInfo: {
          title: "title",
          imageLinks: {
            smallThumbnail: "img-src"
          },
          saleInfo: {
            listPrice: {
              amount: 1,
              currencyCode: "USD"
            },
            country: "EN"
          },
          authors: ["author 1", "author 2"]
        }
      }
    ];

    const component = shallow(
      <CartTotalInfo {...{
        products
      }}/>
    );
    matchSnapshot(component);
  });

  test("Cart total info for cart with a couple different products", () => {
    const products = [
      {
        id: 1,
        amount: 2,
        volumeInfo: {
          title: "title 1",
          imageLinks: {
            smallThumbnail: "img-src"
          },
          saleInfo: {
            listPrice: {
              amount: 1,
              currencyCode: "USD"
            },
            country: "EN"
          },
          authors: ["author 1", "author 2"]
        }
      }, {
        id: 2,
        amount: 4,
        volumeInfo: {
          title: "title 2",
          imageLinks: {
            smallThumbnail: "img-src 2"
          },
          saleInfo: {
            listPrice: {
              amount: 1,
              currencyCode: "USD"
            },
            country: "EN"
          },
          authors: ["author 3", "author 4"]
        }
      }
    ];

    const component = shallow(
      <CartTotalInfo {...{
        products
      }}/>
    );
    matchSnapshot(component);
  });

});