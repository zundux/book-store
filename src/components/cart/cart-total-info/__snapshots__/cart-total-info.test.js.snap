// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart total info Cart total info for cart with a couple copy of product 1`] = `
<h2>
  Total: 
  $2.00
</h2>
`;

exports[`Cart total info Cart total info for cart with a couple different products 1`] = `
<h2>
  Total: 
  $6.00
</h2>
`;

exports[`Cart total info Cart total info for cart with one product 1`] = `
<h2>
  Total: 
  $1.00
</h2>
`;

exports[`Cart total info Cart total info for cart without products 1`] = `""`;
